test/bio_printf_test.c

```diff
@@ -0,0 +1,118 @@
+#include <assert.h>
+#include <stdio.h>
+
+#include <openssl/bio.h>
+#include <test/testutil.h>
+
+#if defined OPENSSL_SYS_WINDOWS
+#define TEST_HAVE_FMEMOPEN 0
+#else
+#define TEST_HAVE_FMEMOPEN 1
+#endif
+
+#define lengthof(a) (sizeof (a) / sizeof *(a))
+#define endof(a) (&(a)[lengthof(a)])
+
+#define TEST_MAXBUF 3
+
+static struct test {
```

Comment:
I would strongly prefer if test data was `static const` so that tests can't modify global state and thereby affect the behavior of later tests. (The vprintf test modifies bufsiz, so the other tests are slightly different depending on `TEST_HAVE_FMEMOPEN`.)